src/test/java/org/shikshalokam/backend/scp/TestScpEntityTypesAndEntitiesCRUDOperations.java

```diff
@@ -0,0 +1,239 @@
+
+        // Call entityTypeUpdation with valid parameters
+        JSONObject requestBody = entityTypeForUpdate("updatedentityTypeValue", "updatedentityTypeLabel", "ACTIVE", "SYSTEM", "false", "string");
+
+        // Call entityTypeUpdation with the created requestBody
+        Response response = entityTypeUpdateRequest(requestBody);
+
+        // Log the status code and response body
+        int statusCode = response.getStatusCode();
+        response.prettyPrint();
+
+        // Validate response code is 201 for a successful update
+        Assert.assertEquals(statusCode, 202, "Status code should be 202");
+
+        logger.info("Ended calling the update entity types API with valid payload.");
+    }
+
+    @Test(dependsOnMethods = "testCreateEntityTypeWithInValidPayload", description = "Verifies the functionality of updating entity types with Invalid payload.")
+    public void testUpdateEntityTypesWithInvalidPayload() {
+        logger.info("Started calling the entity types API with Invalid payload:");
+
+        // Call entityTypeUpdation with invalid parameters
+        JSONObject requestBody = entityTypeForUpdate("__entityTypeValue_", "_updatedentityTypeLabel_", "INACTIVE", "SYSTEM_ADMIN", "true", "array");
+
+        // Call entityTypeUpdation with the created requestBody
+        Response response = entityTypeUpdateRequest(requestBody);
+
+        // Log the status code and response body
+        int statusCode = response.getStatusCode();
+        response.prettyPrint();
+
+        // Validate response code is 400 or 422 for invalid request
+        Assert.assertTrue(statusCode == 400 || statusCode == 422, "Status code should be 400 or 422 for invalid payload");
+
+        logger.info("Ended calling the entity types API with valid payload.");
+    }
+
+    @Test(dependsOnMethods = "testCreateEntityTypeWithEmptyFields", description = "Verifies the functionality of updating user's permission with Invalid payload.")
+    public void testUpdateEntityTypesWithEmptyFields() {
+        logger.info("Started calling the entity types API with empty fields payload:");
+
+        // Call entityTypeUpdation with invalid parameters
+        JSONObject requestBody = entityTypeForUpdate("", "", "", "", "", "");
+
+        // Call entityTypeUpdation with the created requestBody
+        Response response = entityTypeUpdateRequest(requestBody);
+
+        // Log the status code and response body
+        int statusCode = response.getStatusCode();
+        response.prettyPrint();
+
+        // Validate response code for empty fields (usually 400)
+        Assert.assertEquals(statusCode, 400, "Status code should be 400 for empty fields payload");
+
+        logger.info("Ended calling the entity types API with empty fields.");
+    }
+
+    //Method to create request body for entityTypes
+    private JSONObject createEntityType(String value, String label, String status, String type, String data_type, String has_entities) {
```

Comment:
you can do this way so that method will take single argument and code will reduce 
private JSONObject entityTypeForUpdate ( HasMap<String,String > map )
{
JSONObject jsonObject = new JSONObject(map);
}

while calling you can do this  :

HashMap<String, String> map = new HashMap<>();
        map.put("name", "John Doe");
        map.put("age", 30);
        map.put("email", "johndoe@example.com");

entityTypeForUpdate (map);